packages/protobuf/src/encode.ts

```diff
@@ -47,8 +47,15 @@ export function patch(Message: Type, payload: any) {
         }
         // repeated
         if (field.repeated) {
-            const RefMessage = Message.lookupTypeOrEnum(field.type);
-            patched[key] = value.map((v: any) => patch(RefMessage, v));
+            const fieldType = Message.lookupTypeOrEnum(field.type);
+            if (fieldType instanceof Enum) {
+                // NOTE: Enum doesn't require patching both string (keys) and number values are accepted
+                patched[key] = value;
+            } else {
+                patched[key] = value.map((v: any) => patch(fieldType, v));
+            }
+            // NOTE: `fieldType` with fields is a Type otherwise it's a Enum.
+            // patched[key] = fieldType.fields ? value.map((v: any) => patch(fieldType, v)) : value;
```

Comment:
not sure should i use `instaceof Enum` or `fieldType.fields` check

Reply:
`instanceof Enum` is ok here. there will never be need for passing something from outside which would possibly not share the same instance. And this way it is more readable.

Reply:
cool, removing this line then